### PR TITLE
iOS-411 Implement delete server data

### PR DESCRIPTION
**Pull request by ErnestFan**
**What's this do?**
Implement UI for `Delete my SimplyE data`
Remove NYPLSettingsAdvancedVC

**Why are we doing this? (w/ JIRA link if applicable)**
[iOS-411](https://jira.nypl.org/browse/IOS-411)

**How should this be tested? / Do these changes have associated tests?**
1. Log in to NYPL
2. Go to Settings -> Accounts -> NYPL
3. Enable `Sync Bookmarks`
4. Select `Delete my SimplyE data` which is under `Sync Bookmarks`
5. Press `Delete Reading Data` button on the screen that shown up
6. With internet connection, you should be back to the account detail page and an alert of deletion succeed should show up
7. Without internet connection, you should stay on the same page and a `Please try again` alert should show up

**Dependencies for merging? Releasing to production?**
N/A

**Does this include changes that require a new SimplyE/Open eBooks build for QA?**
Will bump build number

**Has the application documentation been updated for these changes?**
N/A

**Did someone actually run this code to verify it works?**
@ErnestFan

### CODE REVIEW

**Comment by ErnestFan** [comment]
@ettore This is mostly done except waiting for some wordings from Marcus. Once I heard back from him, I will need your help on translating it to Italian and I will update the PR after.
Will also bump the build number once everything is ready.

**Comment by ettore** [comment]
@ErnestFan I think this is just for SimplyE and we need to keep the advanced settings VC for open ebooks, or am I wrong?

**Comment by killerpuppytails** [comment]
Per a Slack convo today, Miguel is fine with this being the same on Open eBooks, with the name of the app correctly subbed out.

**Comment by ErnestFan** [comment]
I will keep it as-is then.
@ettore While working on a similar screen for iOS-413, I find that it's better to use tableview instead of the stackview that I am using now. I will update that part of code, will keep you posted once it's ready for review.

**Comment by ErnestFan** [comment]
> I will keep it as-is then. @ettore While working on a similar screen for iOS-413, I find that it's better to use tableview instead of the stackview that I am using now. I will update that part of code, will keep you posted once it's ready for review.

Updated with tableview, still waiting for wordings though. The DeleteServerDataVC was not included in every target, so I fixed that as well.

File: Simplified/Settings/NYPLSettingsDeleteServerDataViewController.swift [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    button.addTarget(self, action: #selector(deleteData), for: .touchUpInside)
+    return button
+  }()
+  
+  private func dataDeletionDescription() -> NSMutableAttributedString {
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.lineSpacing = 1.25
+    paragraphStyle.lineHeightMultiple = 1.25
+    
+    let description = NSLocalizedString("DeleteServerDataDescription",
```
Comment (ettore) [inline Simplified/Settings/NYPLSettingsDeleteServerDataViewController.swift:161]: if you switch to Spanish, is the English translation displayed? I'm afraid "DeleteServerDataDescription" will be displayed instead.... if so you might need to copy the English wording into Spanish and German, sadly.

File: Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import UIKit
+
+/// Display an activity indicator with a message in the center of the screen,
+/// with a dimmed transparent background.
+class NYPLActivityIndicatorMessageViewController: UIViewController {
+  private let spacing: CGFloat = 20.0
+  
+  // MARK: - Life Cycle
+  
+  init(with message: String) {
```
Comment (ettore) [inline Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift:18]: in initializers "with" is typically not used by convention. You can just do `init(message: String)`

File: Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    
+    activityIndicator.startAnimating()
+  }
+  
+  // MARK: - Setup
+  
+  func setupUI() {
```
Comment (ettore) [inline Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift:43]: can this be private?

File: Simplified/Settings/NYPLSettingsAccountDetailViewController.m [hunk 4]
```diff
@@ -1056,18 +1061,38 @@ - (UITableViewCell *)tableView:(__attribute__((unused)) UITableView *)tableView
...
                                reuseIdentifier:nil];
-      cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
       cell.textLabel.font = [UIFont customFontForTextStyle:UIFontTextStyleBody];
-      cell.textLabel.text = NSLocalizedString(@"Advanced", nil);
+      cell.textLabel.text = NSLocalizedString(@"Delete my SimplyE Data", nil);
+      cell.textLabel.textColor = NYPLConfiguration.deleteActionColor;
+      
+      UIImageView *imageView = [self rightArrowImageView];
+      imageView.tintColor = NYPLConfiguration.deleteActionColor;
+      cell.accessoryView = imageView;
```
[the remark this answers is not in the document]
Comment (ErnestFan) [reply Simplified/Settings/NYPLSettingsAccountDetailViewController.m:1074]: The UI requires a red arrow for the delete actions, tintColor does not work on `UITableViewCellAccessoryDisclosureIndicator`

File: Simplified/Settings/NYPLSettingsAccountDetailViewController.m [hunk 5]
```diff
@@ -1056,18 +1064,43 @@ - (UITableViewCell *)tableView:(__attribute__((unused)) UITableView *)tableView
...
       UITableViewCell *cell = [[UITableViewCell alloc]
                                initWithStyle:UITableViewCellStyleDefault
                                reuseIdentifier:nil];
-      cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
       cell.textLabel.font = [UIFont customFontForTextStyle:UIFontTextStyleBody];
-      cell.textLabel.text = NSLocalizedString(@"Advanced", nil);
+#ifdef SIMPLYE
+      cell.textLabel.text = NSLocalizedString(@"Delete my SimplyE Data", nil);
+#else
+      cell.textLabel.text = NSLocalizedString(@"Delete my OpenEBooks Data", nil);
```
Comment (ettore) [inline Simplified/Settings/NYPLSettingsAccountDetailViewController.m:1075]: I think the official spelling is "Open eBooks"

**Review by ettore — APPROVED** [review]

**Pull request merged**